[Automation] native-image run fails for org.xerial:sqlite-jdbc:3.39.2.1

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.39.2.1" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.36.0.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25256364902/job/74056402054
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.assertj/assertj-core/3.22.0/c300c0c6a24559f35fa0bd3a5472dc1edcd0111e/assertj-core-3.22.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.opentest4j/opentest4j/1.2.0/28c11eb91f9b6d8e200631d46e20a7f407f2a046/opentest4j-1.2.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/classes/java/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
'-H:ResourceConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/native-image.properties=META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json' \
'-H:JNIConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/jni-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/jni-config.json' \
'-H:ReflectionConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/reflect-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/reflect-config.json' \
'-H:ResourceConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json' \
-H:ReachabilityMetadataResources@file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/resources/test/META-INF/native-image/reachability-metadata.json+api=META-INF/native-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.xerial.sqlite-jdbc_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.xerial/sqlite-jdbc/3.36.0.3 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=f5fdca99-5bad-19a1-01dc-006e9e19f668 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-25_linux-x64_bin/25.0.0/x64/graalvm-jdk-25.0.3+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature,com.oracle.svm.polyglot.scala.ScalaFeature'
]
========================================================================================================================
GraalVM Native Image: Generating 'org.xerial.sqlite-jdbc_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                    (6.1s @ 0.16GB)
 Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.0.3+9.1
 Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 C compiler: gcc (linux, x86_64, 13.3.0)
 Garbage collector: Serial GC (max heap size: 80% of RAM)
 2 user-specific feature(s):
 - com.oracle.svm.thirdparty.gson.GsonFeature
 - org.graalvm.junit.platform.JUnitPlatformFeature
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:ResourceConfigurationResources' (origin(s): 'META-INF/native-image/org.xerial/sqlite-jdbc/native-image.properties' in 'file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar')
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 12.67GB of memory (75.6% of system memory, $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                   (34.4s @ 0.89GB)
    8,171 types,   9,226 fields, and  35,889 methods found reachable
    2,699 types,     259 fields, and   1,671 methods registered for reflection
       70 types,      92 fields, and     288 methods registered for JNI access
        0 downcalls and 0 upcalls registered for foreign access
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                               (4.3s @ 0.99GB)
[4/8] Parsing methods...      [**]                                                                       (4.2s @ 0.65GB)
[5/8] Inlining methods...     [***]                                                                      (2.3s @ 0.72GB)
[6/8] Compiling methods...    [****]                                                                    (18.6s @ 0.85GB)
[7/8] Laying out methods...   [**]                                                                       (3.2s @ 0.79GB)
[8/8] Creating image...       [**]                                                                       (3.5s @ 0.93GB)
   9.53MB (19.09%) for code area:    22,360 compilation units
  36.57MB (73.24%) for image heap:  214,343 objects and 329 resources
   3.83MB ( 7.67%) for other data
  49.93MB in total image size, 47.91MB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   4.56MB java.base                                           23.78MB byte[] for general heap data
   2.15MB svm.jar (Native Image)                               2.74MB byte[] for code metadata
 558.98kB assertj-core-3.22.0.jar                              2.56MB byte[] for java.lang.String
 481.21kB java.xml                                             1.37MB java.lang.String
 306.29kB sqlite-jdbc-3.39.2.1.jar                             1.02MB java.lang.Class
 207.98kB junit-jupiter-engine-5.9.2.jar                     849.78kB com.oracle.svm.core.hub.DynamicHubCompanion
 120.74kB junit-platform-launcher-1.9.2.jar                  587.52kB heap alignment
 103.49kB java.logging                                       449.73kB java.util.concurrent.ConcurrentHashMap$Node
  99.40kB junit-platform-engine-1.9.2.jar                    304.35kB java.lang.Object[]
  81.79kB junit-4.13.2.jar                                   301.50kB java.util.HashMap$Node
 515.92kB for 25 more packages                                 2.60MB for 2052 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (900B). 5 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        4.8s (6.0% of total time) in 697 GCs | Peak RSS: 1.75GB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests (executable)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'org.xerial.sqlite-jdbc_tests-tests' in 1m 19s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest > createsDatabaseMetadataAndReadsDriverDetails() FAILED

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenExplicitPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenReferenceClassPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteConnectionTest > opensDatabaseFromClasspathResource() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest > initializeExtractsPackagedNativeLibraryResource() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test > pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection() SUCCESSFUL


Failures (1):
  JUnit Jupiter:JDBC3DatabaseMetaDataTest:createsDatabaseMetadataAndReadsDriverDetails()
    MethodSource [className = 'org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest', methodName = 'createsDatabaseMetadataAndReadsDriverDetails', methodParameterTypes = '']
    => java.lang.NullPointerException
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getMethodID(JNIFunctions.java:1828)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetMethodID(JNIFunctions.java:437)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIOnLoadFunctionPointer.invoke(JNILibraryInitializer.java)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNILibraryInitializer.initialize(JNILibraryInitializer.java:119)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibrarySupport.addLibrary(NativeLibrarySupport.java:128)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibrarySupport.addLibrary(NativeLibrarySupport.java:84)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibrary0(NativeLibraries.java:147)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:102)
       java.base@25.0.3/java.lang.ClassLoader.loadLibrary(ClassLoader.java:143)
       java.base@25.0.3/java.lang.Runtime.load0(Runtime.java:767)
       [...]

Test run finished after 31 ms
[         7 containers found      ]
[         0 containers skipped    ]
[         7 containers started    ]
[         0 containers aborted    ]
[         7 containers successful ]
[         0 containers failed     ]
[         6 tests found           ]
[         0 tests skipped         ]
[         6 tests started         ]
[         0 tests aborted         ]
WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::load has been called by org.sqlite.SQLiteJDBCLoader in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 23s
[         5 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.xerial:sqlite-jdbc:3.36.0.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 24s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.39.2.1 failed with exit code 1.
FAILED[native-image run][3.39.2.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.39.2.1" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.36.0.3"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.39.2.1" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.36.0.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25256364902/job/74056402065
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
chability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
'-H:ResourceConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/native-image.properties=META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json' \
'-H:JNIConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/jni-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/jni-config.json' \
'-H:ReflectionConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/reflect-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/reflect-config.json' \
'-H:ResourceConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json' \
-H:ReachabilityMetadataResources@file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/resources/test/META-INF/native-image/reachability-metadata.json+api=META-INF/native-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.xerial.sqlite-jdbc_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.xerial/sqlite-jdbc/3.36.0.3 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=f5fdca99-5bad-19a1-01dc-006e9e19f668 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.xerial.sqlite-jdbc_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.7s @ 0.18GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:ResourceConfigurationResources' (origin(s): 'META-INF/native-image/org.xerial/sqlite-jdbc/native-image.properties' in 'file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar')
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (33.3s @ 0.87GiB)
    8,299 types,   9,281 fields, and  36,328 methods found reachable
      299 types,     107 fields, and     994 methods registered for reflection
       70 types,      92 fields, and     288 methods registered for JNI access
      329 resource accesses registered with 22.25MiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.1s @ 0.97GiB)
[4/8] Parsing methods...      [**]                                                                      (4.1s @ 0.64GiB)
[5/8] Inlining methods...     [***]                                                                     (2.0s @ 0.71GiB)
[6/8] Compiling methods...    [****]                                                                   (19.0s @ 0.72GiB)
[7/8] Laying out methods...   [**]                                                                      (2.8s @ 0.86GiB)
[8/8] Creating image...       [**]                                                                      (3.3s @ 1.00GiB)
   8.43MiB (17.94%) for code area:    22,642 compilation units
  34.88MiB (74.18%) for image heap:  219,058 objects and 197 resources
   3.71MiB ( 7.88%) for other data
  47.02MiB in total image size, 45.06MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
 920.73KiB o.g.n.runtime/com.oracle.svm.core.code             22.26MiB byte[] for embedded resources
 720.99KiB java.base/java.util                                 2.58MiB byte[] for code metadata
 408.35KiB java.base/jdk.internal.classfile.impl               2.53MiB byte[] for string data
 324.03KiB java.base/java.lang                                 1.38MiB java.base/java.lang.String
 242.44KiB java.base/java.util.concurrent                      0.99MiB java.base/java.lang.Class
 228.84KiB java.base/java.text                               907.70KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 224.55KiB java.base/com.sun.crypto.provider                 563.01KiB byte[] for general heap data
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       321.76KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
 155.67KiB java.base/java.io                                 307.78KiB byte[] for reflection metadata
 149.52KiB java.base/java.lang.invoke                        300.00KiB java.base/java.util.HashMap$Node
   4.92MiB for 317 more packages                               2.80MiB for 2065 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (978B). 5 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 FUTR: Use '--future-defaults=all' to prepare for future releases.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.0s (6.3% of total time) in 670 GCs | Peak RSS: 1.62GiB | CPU load: 3.59
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests (executable, 45.06MiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'org.xerial.sqlite-jdbc_tests-tests' in 1m 18s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::load has been called by org.sqlite.SQLiteJDBCLoader in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled

org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest > createsDatabaseMetadataAndReadsDriverDetails() FAILED

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenExplicitPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenReferenceClassPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteConnectionTest > opensDatabaseFromClasspathResource() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest > initializeExtractsPackagedNativeLibraryResource() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test > pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection() SUCCESSFUL


Failures (1):
  JUnit Jupiter:JDBC3DatabaseMetaDataTest:createsDatabaseMetadataAndReadsDriverDetails()
    MethodSource [className = 'org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest', methodName = 'createsDatabaseMetadataAndReadsDriverDetails', methodParameterTypes = '']
    => java.lang.NullPointerException
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getMethodID(JNIFunctions.java:1829)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetMethodID(JNIFunctions.java:438)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIOnLoadFunctionPointer.invoke(JNILibraryInitializer.java)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNILibraryInitializer.initialize(JNILibraryInitializer.java:118)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibrarySupport.addLibrary(NativeLibrarySupport.java:134)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibrarySupport.addLibrary(NativeLibrarySupport.java:90)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibrary0(NativeLibraries.java:148)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:102)
       java.base@25.0.2/java.lang.ClassLoader.loadLibrary(ClassLoader.java:166)
       java.base@25.0.2/java.lang.Runtime.load0(Runtime.java:767)
       [...]

Test run finished after 23 ms
[         7 containers found      ]
[         0 containers skipped    ]
[         7 containers started    ]
[         0 containers aborted    ]
[         7 containers successful ]
[         0 containers failed     ]
[         6 tests found           ]
[         0 tests skipped         ]
[         6 tests started         ]
[         0 tests aborted         ]
[         5 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.


BUILD FAILED in 1m 22s
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.xerial:sqlite-jdbc:3.36.0.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 23s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.39.2.1 failed with exit code 1.
FAILED[native-image run][3.39.2.1][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="3.39.2.1" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.36.0.3"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.39.2.1" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.36.0.3"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25256364902/job/74056402062
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
me/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
'-H:ResourceConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/native-image.properties=META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json' \
'-H:JNIConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/jni-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/jni-config.json' \
'-H:ReflectionConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/reflect-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/reflect-config.json' \
'-H:ResourceConfigurationResources@jar:file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar!/META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json+api=META-INF/native-image/org.xerial/sqlite-jdbc/resource-config.json' \
-H:ReachabilityMetadataResources@file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/resources/test/META-INF/native-image/reachability-metadata.json+api=META-INF/native-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.xerial.sqlite-jdbc_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.xerial/sqlite-jdbc/3.36.0.3 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=8f535ec1-e72b-d03f-506a-e11b61489021 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+10.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.xerial.sqlite-jdbc_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (6.6s @ 0.18GiB)
 Builder configuration:
 - Java version: 25.0.2+10-LTS, vendor version: Oracle GraalVM 25.1.0-dev+10.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:ResourceConfigurationResources' (origin(s): 'META-INF/native-image/org.xerial/sqlite-jdbc/native-image.properties' in 'file:///home/runner/.gradle/caches/modules-2/files-2.1/org.xerial/sqlite-jdbc/3.39.2.1/cec756738414b14561dae0caf0ccb988e0529954/sqlite-jdbc-3.39.2.1.jar')
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [*****]                                                                  (36.1s @ 0.96GiB)
    8,827 types,  10,060 fields, and  39,310 methods found reachable
      427 types,     107 fields, and   1,185 methods registered for reflection
       70 types,      92 fields, and     288 methods registered for JNI access
      331 resource accesses registered with 22.25MiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (4.2s @ 1.07GiB)
[4/8] Parsing methods...      [**]                                                                      (4.5s @ 0.79GiB)
[5/8] Inlining methods...     [***]                                                                     (2.0s @ 0.83GiB)
[6/8] Compiling methods...    [****]                                                                   (21.0s @ 0.85GiB)
[7/8] Laying out methods...   [**]                                                                      (3.1s @ 1.03GiB)
[8/8] Creating image...       [**]                                                                      (3.8s @ 1.18GiB)
   9.53MiB (18.81%) for code area:    24,924 compilation units
  37.19MiB (73.36%) for image heap:  256,417 objects and 199 resources
   3.97MiB ( 7.83%) for other data
  50.69MiB in total image size, 48.56MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.03MiB o.g.n.runtime/com.oracle.svm.core.code             22.26MiB byte[] for embedded resources
 728.14KiB java.base/java.util                                 2.93MiB byte[] for code metadata
 408.35KiB java.base/jdk.internal.classfile.impl               2.81MiB byte[] for string data
 332.81KiB java.base/java.lang                                 1.50MiB java.base/java.lang.String
 267.01KiB java.base/com.sun.crypto.provider                   1.06MiB java.base/java.lang.Class
 242.65KiB java.base/java.util.concurrent                    965.45KiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 231.41KiB java.base/java.text                               921.84KiB byte[] for general heap data
 223.93KiB org.assertj.core.internal.bytebuddy.jar.asm       498.20KiB heap alignment
 192.84KiB java.base/sun.security.provider                   388.85KiB java.base/java.lang.Object[]
 173.16KiB java.base/sun.security.x509                       375.77KiB byte[] for reflection metadata
   5.77MiB for 327 more packages                               3.55MiB for 2212 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 17 component(s) is embedded in binary (976B). 5 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       5.9s (6.9% of total time) in 786 GCs | Peak RSS: 1.80GiB | CPU load: 3.61
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests (executable, 48.56MiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'org.xerial.sqlite-jdbc_tests-tests' in 1m 24s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest > createsDatabaseMetadataAndReadsDriverDetails() FAILED

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenExplicitPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.ResourceFinderTest > returnsNullWhenReferenceClassPackageResourceIsAbsent() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteConnectionTest > opensDatabaseFromClasspathResource() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLiteJDBCLoaderTest > initializeExtractsPackagedNativeLibraryResource() SUCCESSFUL

org_xerial.sqlite_jdbc.SQLitePooledConnectionAnonymous1Test > pooledConnectionProxyDelegatesConnectionMethodsToPhysicalConnection() SUCCESSFUL


Failures (1):
  JUnit Jupiter:JDBC3DatabaseMetaDataTest:createsDatabaseMetadataAndReadsDriverDetails()
    MethodSource [className = 'org_xerial.sqlite_jdbc.JDBC3DatabaseMetaDataTest', methodName = 'createsDatabaseMetadataAndReadsDriverDetails', methodParameterTypes = '']
    => java.lang.NullPointerException
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions$Support.getMethodID(JNIFunctions.java:1829)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.GetMethodID(JNIFunctions.java:438)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNIOnLoadFunctionPointer.invoke(JNILibraryInitializer.java)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.JNILibraryInitializer.initialize(JNILibraryInitializer.java:118)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibrarySupport.addLibrary(NativeLibrarySupport.java:134)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibrarySupport.addLibrary(NativeLibrarySupport.java:90)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibrary0(NativeLibraries.java:148)
       org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk.NativeLibraries.loadLibraryAbsolute(NativeLibraries.java:102)
       java.base@25.0.2/java.lang.ClassLoader.loadLibrary(ClassLoader.java:166)
       java.base@25.0.2/java.lang.Runtime.load0(Runtime.java:767)
       [...]

Test run finished after 25 ms
[         7 containers found      ]
[         0 containers skipped    ]
[         7 containers started    ]
[         0 containers aborted    ]
[         7 containers successful ]
[         0 containers failed     ]
[         6 tests found           ]
[         0 tests skipped         ]
[         6 tests started         ]
[         0 tests aborted         ]
[         5 tests successful      ]
[         1 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date
WARNING: A restricted method in java.lang.System has been called
WARNING: java.lang.System::load has been called by org.sqlite.SQLiteJDBCLoader in an unnamed module (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests)
WARNING: Use --enable-native-access=ALL-UNNAMED to avoid a warning for callers in this module
WARNING: Restricted methods will be blocked in a future release unless native access is enabled


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.xerial/sqlite-jdbc/3.36.0.3/build/native/nativeTestCompile/org.xerial.sqlite-jdbc_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 28s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.xerial:sqlite-jdbc:3.36.0.3 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 29s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 3.39.2.1 failed with exit code 1.
FAILED[native-image run][3.39.2.1][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="3.39.2.1" ./gradlew clean test -Pcoordinates="org.xerial:sqlite-jdbc:3.36.0.3"]
```

--- a/metadata/org.xerial/sqlite-jdbc/3.39.2.1/reachability-metadata.json
+++ b/metadata/org.xerial/sqlite-jdbc/3.39.2.1/reachability-metadata.json
@@ -1,0 +1,266 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "java.lang.Throwable",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "toString",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.javax.SQLitePooledConnection$1"
+      },
+      "type" : "java.sql.Connection",
+      "methods" : [
+        {
+          "name" : "createStatement",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "getAutoCommit",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "isClosed",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.Collation",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "xCompare",
+          "parameterTypes" : [
+            "java.lang.String",
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.Function",
+      "jniAccessible" : true,
+      "fields" : [
+        {
+          "name" : "args"
+        },
+        {
+          "name" : "context"
+        },
+        {
+          "name" : "value"
+        }
+      ],
+      "methods" : [
+        {
+          "name" : "xFunc",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.Function$Aggregate",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "clone",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "xFinal",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "xStep",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.Function$Window",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "xInverse",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "xValue",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.ProgressHandler",
+      "jniAccessible" : true
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.core.DB",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "throwex",
+          "parameterTypes" : [ ]
+        },
+        {
+          "name" : "throwex",
+          "parameterTypes" : [
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.core.DB$ProgressObserver",
+      "jniAccessible" : true,
+      "methods" : [
+        {
+          "name" : "progress",
+          "parameterTypes" : [
+            "int",
+            "int"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "org.sqlite.core.NativeDB",
+      "jniAccessible" : true,
+      "fields" : [
+        {
+          "name" : "pointer"
+        }
+      ],
+      "methods" : [
+        {
+          "name" : "stringToUtf8ByteArray",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        },
+        {
+          "name" : "throwex",
+          "parameterTypes" : [
+            "java.lang.String"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.core.NativeDB"
+      },
+      "type" : "org.sqlite.core.NativeDB",
+      "jniAccessible" : true,
+      "fields" : [
+        {
+          "name" : "busyHandler"
+        },
+        {
+          "name" : "commitListener"
+        },
+        {
+          "name" : "progressHandler"
+        },
+        {
+          "name" : "updateListener"
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "sun.security.provider.NativePRNG",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [
+            "java.security.SecureRandomParameters"
+          ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "type" : "sun.security.provider.SHA",
+      "methods" : [
+        {
+          "name" : "<init>",
+          "parameterTypes" : [ ]
+        }
+      ]
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.javax.SQLitePooledConnection"
+      },
+      "type" : {
+        "proxy" : [
+          "java.sql.Connection"
+        ]
+      }
+    }
+  ],
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.util.ResourceFinder"
+      },
+      "glob" : "/org_xerial/sqlite_jdbc/resource_finder_missing_resource.txt"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "glob" : "META-INF/maven/org.xerial/sqlite-jdbc/pom.properties"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.SQLiteJDBCLoader"
+      },
+      "glob" : "org/sqlite/native/Linux/x86_64/libsqlitejdbc.so"
+    },
+    {
+      "condition" : {
+        "typeReached" : "org.sqlite.jdbc3.JDBC3DatabaseMetaData"
+      },
+      "glob" : "sqlite-jdbc.properties"
+    }
+  ]
+}

--- a/metadata/org.xerial/sqlite-jdbc/index.json
+++ b/metadata/org.xerial/sqlite-jdbc/index.json
@@ -1,6 +1,21 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.39.2.1",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/xerial/sqlite-jdbc/$version$/sqlite-jdbc-$version$-sources.jar",
+    "repository-url" : "https://github.com/xerial/sqlite-jdbc",
+    "test-code-url" : "https://github.com/xerial/sqlite-jdbc/tree/$version$/src/test",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/xerial/sqlite-jdbc/$version$/sqlite-jdbc-$version$-javadoc.jar",
+    "description" : "SQLite JDBC is a JDBC driver for accessing and creating SQLite database files from Java applications. It packages native SQLite libraries for major operating systems into a single JAR so applications can use SQLite without separate native setup.",
+    "test-version" : "3.36.0.3",
+    "tested-versions" : [
+      "3.39.2.1"
+    ],
+    "allowed-packages" : [
+      "org.sqlite"
+    ]
+  },
+  {
     "metadata-version" : "3.36.0.3",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/xerial/sqlite-jdbc/3.36.0.3/sqlite-jdbc-3.36.0.3-sources.jar",
     "repository-url" : "https://github.com/xerial/sqlite-jdbc",

--- a/stats/org.xerial/sqlite-jdbc/3.39.2.1/stats.json
+++ b/stats/org.xerial/sqlite-jdbc/3.39.2.1/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "3.39.2.1",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 3,
+          "totalCalls" : 3
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 5,
+          "totalCalls" : 5
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 8,
+      "totalCalls" : 8
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 5542,
+        "missed" : 14549,
+        "ratio" : 0.275845,
+        "total" : 20091
+      },
+      "line" : {
+        "covered" : 1212,
+        "missed" : 3363,
+        "ratio" : 0.264918,
+        "total" : 4575
+      },
+      "method" : {
+        "covered" : 256,
+        "missed" : 1079,
+        "ratio" : 0.19176,
+        "total" : 1335
+      }
+    }
+  } ]
+}

